src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/MarlinInterface.java

```diff
@@ -208,46 +220,16 @@ public boolean getIsBusy() {
 		return busyCount<=0;
 	}
 
-	
-	protected JToolBar getToolBar() {
-		JToolBar bar = new JToolBar();
-		bar.setRollover(true);
-
-		bESTOP.setFont(getFont().deriveFont(Font.BOLD));
-		bESTOP.setForeground(Color.RED);
-
-		bESTOP.addActionListener((e) -> sendESTOP() );
-		bar.add(bESTOP);
-		//bar.addSeparator();
-		
-		updateButtonAccess();
-
-		return bar;
-	}
-
-	private void sendESTOP() {
+	public void sendESTOP() {
+		logger.warn("Emergency stop");
 		chatInterface.sendCommand("M112");
 		chatInterface.sendCommand("M112");
 		chatInterface.sendCommand("M112");
 	}
 
-	private void updateButtonAccess() {
-		boolean isConnected = chatInterface.getIsConnected();
-
-		bESTOP.setEnabled(isConnected);
-	}
-
-	public void setNetworkSession(NetworkSession session) {
-		chatInterface.setNetworkSession(session);
```

Comment:
chatInterface still exists.  so how does it get the right NetworkSession?

Reply:
An event is emitted from the `ConnectionButton` and `TextInterfaceToNetworkSession` is listening for it here:
https://github.com/MarginallyClever/Makelangelo-software/blob/138fc0f42d29dd9d2f76a1588aece8e8b21c3454/src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/TextInterfaceToNetworkSession.java#L48

Reply:
Ok, when is the button access updated if the updateButtonAccess method is deleted?

Reply:
In PlotterControls when the connection is opened
https://github.com/coliss86/Makelangelo-software/blob/60990f77f3f660c5fb058b25e5d6a9f6f471e666/src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/PlotterControls.java#L228